### ISSUE TITLE
BinaryPattern - should be formatted like BinaryExpression

### ISSUE THREAD

**Issue by belav** [belav/csharpier#349]
BinaryPatterns are essentially the same as a BinaryExpression and should be printed the same.
They have a Left, OperatorToken and Right.
BinaryPatterns are not currently indented, and things like IsPattern do the indentation, which I don't believe is the way it should be handled.
If we keep IsPattern indenting, then SwitchExpression will also need to indent.

See #37, possibly do this at the same time.

**Comment by belav** [comment]
Mixed example
```c#
if (
      nestedExpression.Children[j]
          is IntermediateToken cSharpToken && cSharpToken.IsCSharp
  ) { }

// should be

if (
      nestedExpression.Children[j]
          is IntermediateToken cSharpToken
      && cSharpToken.IsCSharp
  ) { }
```

**Comment by belav** [comment]
Closes this, it can be handled in #154
